core/include/gnuradio-4.0/Tag.hpp

```diff
@@ -109,6 +106,21 @@ struct alignas(hardware_constructive_interference_size) Tag {
     void insert_or_assign(const std::string& key, const pmtv::pmt& value) { map[key] = value; }
 };
 
+using Tag              = BasicTag<std::size_t>;
+using RelativeIndexTag = BasicTag<std::make_signed_t<std::size_t>>;
+
+constexpr std::make_signed_t<std::size_t> tagIndexDifference(Tag::index_type x, Tag::index_type y) {
+    // assuming that tag indices and stream position are never further apart than half the range of the index type,
+    // assume integer overflow if they are.
```

Comment:
This code only makes sense if we decide to care about std::size_t overflow, otherwise it should be simplified.